scripts/install_prereqs.py

```diff
@@ -0,0 +1,453 @@
+            print('ccache requested but is already installed. Skipping installation...')
+        elif is_linux_os() or is_mac_os():
+            system_packages_to_install.append('ccache')
+
+    if system_packages_to_install:
+        install_system_packages(system_packages_to_install)
+
+
+def step_install_python_packages(args):
+    """ Install python packages (dependencies)
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments
+    """
+    # Check for packages installed in this python environment
+    installed_packages = {pkg.key for pkg in pkg_resources.working_set}
+
+    # Get the list of packages we intend to install
+    requirements_file_path = get_pip_requirements_file_path()
+    packages_to_install = None
+    with open(requirements_file_path, 'r') as requirements_file:
+        packages_to_install = {line.strip('\n') for line in requirements_file.readlines()}
+
+    # Check if there is any work to do by comparing these lists
+    if packages_to_install and not (packages_to_install-installed_packages):
+        print('Skipping python packages installation because following '\
+               'python packages are already installed: \n {0}'.format(' '.join(packages_to_install)))
+        return
+
+    install_python_packages()
+
+
+def step_install_cpp_packages(args):
+    """ Install C++ packages (dependencies)
+
+    Installation supported only via vcpkg as system package manager can affect
+    system wide packages. User should handle those themselves
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments
+    """
+    if not args.no_vcpkg:
+        # Check if vcpkg is already installed
+        found_vcpkg_executable = os.path.exists(get_vcpkg_executable_file_path())
+
+        # Install packages and only install vcpkg if its not built already
+        install_packages_with_vcpkg(args.arch, build_vcpkg=not found_vcpkg_executable)
+
+    else:
+        # Do not install dependencies automatically as we want users to have control
+        # on their system level package installations
+        print('Please install packages listed in {0}'\
+              'with your system package manager'.format(get_vcpkg_response_file_path(get_vcpkg_triplet(args.arch))))
+
+
+def main():
+    args = parse_cmdline_args()
+
+    step_install_system_packages(args)
+    step_install_python_packages(args)
+    step_install_cpp_packages(args)
+
+
+def parse_cmdline_args():
+    parser = argparse.ArgumentParser(description='Install Prerequisites for building cpp sdk')
+    parser.add_argument('-A', '--arch', default='x64', help='Platform architecture (x64, x86)')
+    parser.add_argument('-G', '--generator', help='Build generator to use (install ninja if needed)')
+    parser.add_argument('-c', '--ccache', action='store_true', help='Install ccache using system package manager'\
+                                                                    '(only supported on mac and linux)')
```

Comment:
So I tried to keep it as simple as possible but this is the bare minimum required for prereqs. The reasons are,
- architecture determines the "vcpkg triplet". Basically vcpkg build step needs to know which architecture/platform it is building for.
-  --generator and --ccache : These were included only to support Ninja and ccache. If we dont plan to use Ninja, we can get rid of "--generator". The prereqs script will install these two tools via a system package manager because they are not available in vcpkg. 

Having said this, the script can run without any command line flags whatsoever and it will work. (it assumes default arch of x64 and it just won't install ninja and ccache)

Reply:
architecture: The idea of breaking up the script into prereq / build is that you install_prereqs once per system, and then can build multiple times for different architectures. This means that if vcpkg has to be run separately for each architecture, I think we should either a) install it for all architectures in this prereqs script   or b) run vcpkg in the build script.

generator: As previously commented, let's get rid of Ninja and just use the default.

ccache: Just always install it when it's available (i.e. not on Windows).